library/ssl_srv.c

```diff
@@ -298,13 +298,13 @@ static int ssl_parse_signature_algorithms_ext( mbedtls_ssl_context *ssl,
         {
             mbedtls_ssl_sig_hash_set_add( &ssl->handshake->hash_algs, sig_cur, md_cur );
             MBEDTLS_SSL_DEBUG_MSG( 3, ( "client hello v3, signature_algorithm ext:"
-                                        " match sig %d and hash %d",
+                                        " match sig %u and hash %u",
```

Comment:
`mbedtls_pk_type_t` and `mbedtls_md_type_t` are enums. In C, the size of an enumerated type is implementation-defined and can be as large as the compiler wants. ([Cppreference says otherwise, but that's in C++](https://en.cppreference.com/w/cpp/language/enum): in [C99](http://www.iso-9899.info/n1256.html#6.7.2.2p4) and [C11](http://www.iso-9899.info/n1570.html#6.7.2.2p4) there is no upper limit). Therefore `sig_cur` and `md_cur` need to be cast. We know that the values are small positive values, so a cast to `unsigned` is fine.

Same with `md_cur` in the else branch just below.